registration/include/pcl/registration/ndt.h

```diff
@@ -288,6 +324,8 @@ namespace pcl
       double
       updateDerivatives (Eigen::Matrix<double, 6, 1> &score_gradient,
                          Eigen::Matrix<double, 6, 6> &hessian,
+                         const Eigen::Matrix<float, 4, 6>& point_gradient,
```

Comment:
Can we prevent a complete breakage of API here? Is it possible to instead provide a migration path:
1. deprecate the `point_jacobian_` and `point_hessian_`
2. deprecate the original function declaration, make it check `num_threads_` and call the new implementation
3. default the new implementation

Reply:
1. I reverted ```point_jacobian_``` and ```point_hessian_``` and added deprecation flags to them
2. I reverted their declaration and definitions